src/base/bittorrent/infohash.h

```diff
@@ -38,15 +43,45 @@ using SHA256Hash = Digest32<256>;
 
 namespace BitTorrent
 {
-    class InfoHash : public SHA1Hash
+    class InfoHash;
+
+    class TorrentID : public Digest32<160>
     {
     public:
-        using SHA1Hash::SHA1Hash;
+        using BaseType = Digest32<160>;
```

Comment:
Do you intend to make it `public`? maybe is it better to hide it in `private`?

Reply:
>Sorry, what's the point to hide it?

To only disclosure what are needed to the public.

Another thing, you use different names for `TorrentID::BaseType` and `InfoHash::UnderlyingType` while they could be both named `UnderlyingType` for me. If they are placed in `private` then I wouldn't mention this.

Reply:
> you use different names for `TorrentID::BaseType` and `InfoHash::UnderlyingType`

they mean different things, in case you haven't noticed

Reply:
> To only disclosure what are needed to the public..

I disagree. Only the implementation details must be private. There is no point in hiding something that can be safely used from the outside.

Reply:
>There is no point in hiding something that can be safely used from the outside.

OK, it seems to have some valid usage.

However there is one thing that I must note: It seems C++ lang is fixed on coining "underlying type" only for enum types, they even added a `std::to_underlying()` function in C++23 that only works for a `enum`. It would be better to stay away and pick another alias (e.g. `HashType` or `InternalType`) to avoid such confusion in the future.

Reply:
> It would be better to stay away and pick another alias (e.g. `HashType` or `InternalType`) to avoid such confusion in the future.

Agree.

Reply:
`HashType` sounds too narrow. Then we'll have to invent new names in every case like this (when we need to wrap some other type). `InternalType` gives the impression that we are revealing some implementation details. Maybe something like `WrappedType`?

Reply:
>Maybe something like WrappedType?

Let's settle on that for now, I can't find more suitable names either.

Reply:
> > Maybe something like WrappedType?
> 
> Let's settle on that for now, I can't find more suitable names either.

Ok. Changed.